src/Roave/SecurityAdvisories/Version.php

```diff
@@ -37,85 +38,90 @@ public static function fromString(string $version) : self
             throw new \InvalidArgumentException(sprintf('Given version "%s" is not a valid version string', $version));
         }
 
-        return new self(self::removeTrailingZeroes(array_map('intval', explode('.', $version))));
+        return new self($version);
```

Comment:
we actually need version to be intact and even add zeroes when comparing versions. This way `version_compare` works correctly

Reply:
Hmm, there was a reason why I didn't use `version_compare()` in the past, but can't remember which.

What happens if you compare `1.0.0` with `1.0` now?

Reply:
answer: it will return false.

well, I checked semver code and also this repository - https://github.com/nikolaposa/version
both of them are using `version_compare()`, but it is tricky, here is example of equal to comparison
 https://3v4l.org/Hjth6

Reply:
That is indeed something we'll need to fix then. `1.0` and `1.0.0` are the same, for our purposes.

Reply:
As discussed above, the `VersionParser` cannot be used here, because it cannot cope with valid constraints such as `'0.1.2.3.4'`. Also, normalisation should happen at instantiation to ease comparison of equivalent objects.